packages/firebase_dynamic_links/android/build.gradle

```diff
@@ -34,8 +34,33 @@ android {
     dependencies {
         api 'com.google.firebase:firebase-dynamic-links:16.1.8'
         implementation 'com.google.firebase:firebase-common:16.1.0'
-        implementation 'androidx.annotation:annotation:1.0.0'
     }
 }
 
 apply from: file("./user-agent.gradle")
+
+// TODO(bparrishMines): Remove this hack once androidx.lifecycle is included on stable. https://github.com/flutter/flutter/issues/42348
+afterEvaluate {
+    def containsEmbeddingDependencies = false
+    for (def configuration : configurations.all) {
+        for (def dependency : configuration.dependencies) {
+            if (dependency.group == 'io.flutter' &&
+                    dependency.name.startsWith('flutter_embedding') &&
+                    dependency.isTransitive())
+            {
+                containsEmbeddingDependencies = true
+                break
+            }
+        }
+    }
+    if (!containsEmbeddingDependencies) {
+        android {
+            dependencies {
+                def lifecycle_version = "1.1.1"
+                compileOnly "android.arch.lifecycle:runtime:$lifecycle_version"
```

Comment:
are there existing androidx transitive dependencies? (since there's an androidx annotation already) Is this a downgrade?

Reply:
These use the latest instruction from the migration wiki: https://github.com/flutter/flutter/wiki/Experimental:-Create-Flutter-Plugin.

Quoting: https://github.com/flutter/plugins/pull/2221#issue-329949430
`We introduced a constraint that the app uses the plugin has to be migrated to androidx.
This patch removes the constraint.`

Reply:
@mklim @blasten we had this conversation separately too. I guess we're saying mixing is ok and we're just letting plugins build as aars as fallbacks?